### PR TITLE
Damlc test color output

### PR DESCRIPTION
**Pull request by akalburgi**
This PR adds a new command line option to enable colored output for the tests ` da-hs-damlc-app -- test --color ` 
This PR addresses issue #82  
### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by akalburgi** [comment]
Screen Shot 2019-04-19 at 2 57 52 PM

**Comment by akalburgi** [comment]
image

File: daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc.hs [hunk 1]
```diff
@@ -98,12 +98,11 @@ cmdTest numProcessors =
...
       <$> many inputFileOpt
+      <*> colorOutPut
       <*> junitOutput
       <*> optionsParser numProcessors optPackageName
-    junitOutput = optional $ strOption $
-        long "junit" <>
-        metavar "FILENAME" <>
-        help "Filename of JUnit output file"
+    junitOutput = optional $ strOption $ long "junit" <> metavar "FILENAME" <> help "Filename of JUnit output file"
+    colorOutPut = switch $ long "colored" <> help "Colored test results"
```
Comment (neil-da) [inline daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc.hs]: `--color` is probably a sufficient flag name. And I'd name the function `colorOutput`.

File: daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc/Test.hs [hunk 2]
```diff
@@ -33,8 +33,8 @@ import qualified Text.XML.Light as XML
 
 
 -- | Test a DAML file.
-execTest :: [FilePath] -> Maybe FilePath -> Compiler.Options -> IO ()
-execTest inFiles mbJUnitOutput cliOptions = do
+execTest :: [FilePath] -> Bool -> Maybe FilePath -> Compiler.Options -> IO ()
```
Comment (neil-da) [inline daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc/Test.hs]: We try and avoid `Bool` arguments. See elsewhere for where we've lifted these into a type.

Comment (hurryabit) [reply daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc/Test.hs]: @akalburgi That's the thing because of which I once put you on a PR out of the blue. If you grep for a line containing both `newtype` and `Bool`, you'll find a good way to deal with this.

Comment (akalburgi) [reply daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc/Test.hs]: @martin-drhu-da  i think it is `type Vat = Bool ` no ?

Comment (hurryabit) [reply daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc/Test.hs]: @akalburgi I meant something along
https://github.com/digital-asset/daml/blob/0b87c4ceaec14809fb0de37e1af950dd74540717/daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc.hs#L224

Comment (akalburgi) [reply daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc/Test.hs]: Ok that makes sense after reading a little bit about newtype vs type keywords

**Comment by akalburgi** [comment]
Screen Shot 2019-04-23 at 1 57 57 PM

File: daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc.hs [hunk 3]
```diff
@@ -98,12 +98,11 @@ cmdTest numProcessors =
   where
     cmd = execTest
       <$> many inputFileOpt
+      <*> fmap (\x -> ColorTestResults x) colorOutput
```
Comment (hurryabit) [inline daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc.hs]:
```suggestion
      <*> fmap ColorTestResults colorOutput
```

File: daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc.hs [hunk 4]
```diff
@@ -98,12 +98,11 @@ cmdTest numProcessors =
...
       <*> junitOutput
       <*> optionsParser numProcessors optPackageName
-    junitOutput = optional $ strOption $
-        long "junit" <>
-        metavar "FILENAME" <>
-        help "Filename of JUnit output file"
+    junitOutput = optional $ strOption $ long "junit" <> metavar "FILENAME" <> help "Filename of JUnit output file"
```
Comment (hurryabit) [inline daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc.hs]: Why did you reformat this?

File: daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc.hs [hunk 5]
```diff
@@ -98,12 +98,11 @@ cmdTest numProcessors =
...
+    colorOutput = switch $ long "color" <> help "Colored test results"
```
Comment (hurryabit) [inline daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc.hs]: "color" without the "u" looks so weird. ;)

File: daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc/Test.hs [hunk 6]
```diff
@@ -32,9 +35,11 @@ import System.FilePath
 import qualified Text.XML.Light as XML
 
 
+newtype ColorTestResults = ColorTestResults{color :: Bool}
```
Comment (hurryabit) [inline daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc/Test.hs]: I'd prefer if the field was called `getColorTestResults` or `unColorTestResults`.

File: daml-foundations/daml-tools/da-hs-daml-cli/tests/DamlcTest.hs [hunk 7]
```diff
@@ -8,8 +8,8 @@ module DamlcTest
 import Control.Exception
 import qualified Data.Text.Extended as T
 import System.Environment
-import System.Exit
 import System.IO.Extra
+import System.Exit
```
Comment (hurryabit) [inline daml-foundations/daml-tools/da-hs-daml-cli/tests/DamlcTest.hs]: Why did you move this?

File: daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc.hs (hunk 4 above)
Comment (akalburgi) [reply daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc.hs]: Felt it was more readable in a line vs having it in multi line

File: daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc/Test.hs (hunk 6 above)
Comment (akalburgi) [reply daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc/Test.hs]: Cool looks good to me too, will make the change
